### DISCUSSION TITLE
图元聚类操作

### DISCUSSION THREAD

**Discussion by MangMax** [zouyaoji/vue-cesium#390] [Ideas]
大佬有没有兴趣做图元的聚类操作，我看到有人已经做了，但苦于本人技术太菜整合不到vue-cesium中
https://github.com/tingyuxuan2302/cesium-complete/blob/github/src/utils/cesiumCtrl/primitiveCluster.js
https://github.com/tingyuxuan2302/cesium-complete/blob/github/src/views/mark/primitive.vue

**Comment by zouyaoji** [comment]
最近有点忙，先mark一下啊，过段时间弄进去。

**Comment by MangMax** [comment]
感谢大佬！

**Reply by zouyaoji** [reply]
我看他这个聚合貌似要闪烁啊？

**Reply by MangMax** [reply]
> 我看他这个聚合貌似要闪烁啊？

第一次加载会闪烁吧，后面好像就正常了，我使用官方实体聚类的话数据量大好像也会有类似闪烁的现象

**Reply by zouyaoji** [reply]
https://zouyaoji.top/vue-cesium/#/zh-CN/component/primitives/vc-primitive-cluster

你空了看看有木有问题。

**Reply by MangMax** [reply]
> https://zouyaoji.top/vue-cesium/#/zh-CN/component/primitives/vc-primitive-cluster
> 
> 你空了看看有木有问题。

npm上貌似还木有

**Reply by zouyaoji** [reply]
还没发布呢，可以先简单测一下，有毛病可以我可以发布前改了。

还有个别人反馈的bug没看，解决了一起发。应该就这两天了。

**Comment by zouyaoji** [comment]
3.1.9版本已加入

**Reply by MangMax** [reply]
升级后出现两个问题
![image](https://github.com/zouyaoji/vue-cesium/assets/68525017/dc6f11c4-31c4-4d5d-bafc-430204f123de)

![image](https://github.com/zouyaoji/vue-cesium/assets/68525017/27ff4945-20e1-4a6e-b37f-eb3d06cd98b8)

**Reply by zouyaoji** [reply]
升级一下cesium版本

**Reply by MangMax** [reply]
> 升级一下cesium版本

由于要使用超图的插件所以不能超过1.90版本，否则插件不兼容无法使用

**Reply by zouyaoji** [reply]
搞个简单的复现的例子我看看具体是哪儿导致的错误。

**Reply by MangMax** [reply]
> 搞个简单的复现的例子我看看具体是哪儿导致的错误。
这个问题好像使用影像图层就会出现，我使用vite创建ts vue项目，pnpm安装vue-cesium，引入cesium，组件只加载viewer和百度地图，报错就会出现
经过测试，104版本开始错误消失

![image](https://github.com/zouyaoji/vue-cesium/assets/68525017/372d0946-0a30-4211-bff1-7ea57760ce2b)
![image](https://github.com/zouyaoji/vue-cesium/assets/68525017/0ac1e0b9-9306-4bc7-88df-3e215752e73f)
![image](https://github.com/zouyaoji/vue-cesium/assets/68525017/e252b4af-5914-4bb1-ab1d-9a1a041b75d8)
![image](https://github.com/zouyaoji/vue-cesium/assets/68525017/8fb998a5-ebb4-49db-ab28-f12d5c4b923a)
![image](https://github.com/zouyaoji/vue-cesium/assets/68525017/cb98089a-948b-4787-91c9-44d33dac56df)

如果需要的话，我将这个测试例子传到github，codepan使用cdn无法使用最新版的vue-cesium，3.1.6以上版本都会报错
![image](https://github.com/zouyaoji/vue-cesium/assets/68525017/f17a5072-f103-4bb5-bdf9-cc415364703c)

**Reply by zouyaoji** [reply]
已修复哈 下个版本就没问题了

**Reply by MangMax** [reply]
> 已修复哈 下个版本就没问题了
十分感谢！

**Reply by MangMax** [reply]
鼠标事件貌似无法响应，我i尝试官网的例子，点击可以触发，mouseover没有反应，我本地工程click和over、out都没效果。
https://codepen.io/1061875478/pen/QWJGazN?editors=1111

**Comment by zmcode** [comment]
例子的聚合量都很少, 我这边的业务都是上万的, 有人遇到过吗? 而且聚合的数字一直在变,哪怕我不缩放

**Reply by zouyaoji** [reply]
聚合是根据屏幕像素来做的，不缩放改了相机倾角这些一样有影响。

**Reply by zmcode** [reply]
> 聚合是根据屏幕像素来做的，不缩放改了相机倾角这些一样有影响。

有没有什么办法能实现只渲染可视区域的聚合，还有这个聚合数据删除了，内存也没释放，我都是按照它源码中的destroy方法调用不行，随时图元没显示了，但是内存还占用，而已聚合计算都会删除之前的数据在生成新的，那些数据额内存一样没释放
